Ska/Sun/sun.py

```diff
@@ -2,14 +2,29 @@
 """
 Utility for calculating sun position and pitch angle.
 """
+from pathlib import Path
+import numpy as np
+from math import cos, sin, acos, atan2, asin, pi, radians, degrees
+from astropy.table import Table
 
 from Quaternion import Quat
 from chandra_aca.transform import radec_to_eci
 from Chandra.Time import DateTime
-from math import cos, sin, acos, atan2, asin, pi, radians, degrees
-import numpy as np
 import Ska.quatutil
 
+ROLL_TABLE = Table.read(str(Path(__file__).parent / 'data' / 'pitch_rolldev.csv'),
+                        format='ascii.basic',
+                        guess=False, delimiter=',')
+
+
+def allowed_rolldev(pitch):
+    """Get allowed roll deviation (off-nominal roll) for the given ``pitch``.
+    :param pitch: Sun pitch angle (deg)
+    :returns: Roll deviation (deg)
+    """
+    idx = np.searchsorted(ROLL_TABLE['pitch'], pitch, side='right')
+    return ROLL_TABLE['rolldev'][idx - 1]
```

Comment:
There is a problem here with the `idx - 1`:
```
In [1]: import Ska.Sun
Sk
In [2]: Ska.Sun.__file__
Out[2]: '/Users/aldcroft/git/Ska.Sun/Ska/Sun/__init__.py'

In [3]: Ska.Sun.allowed_rolldev(40)
Out[3]: 18.7487718079019  <== corresponds to the [-1] entry
```

Reply:
Good catch.  I did not test that the previous routine was working for the corner cases.